plugins/modules/discovery_intent.py

```diff
@@ -364,6 +367,17 @@ def validate_input(self):
                              'default': 1}
         }
 
+        if state == "merged":
+            discovery_spec["ip_address_list"] = {'type': 'list', 'required': True,
+                                                 'elements': 'str'}
+            discovery_spec["discovery_type"] = {'type': 'str', 'required': True}
+
+        if (self.config[0].get("delete_all") is True and state == "deleted"):
```

Comment:
Line 370 we checked state is merged. If state is merged, then we do comparision in line 375 as well. Instead can we write below?
```
        if state == "merged":
            discovery_spec["ip_address_list"] = {'type': 'list', 'required': True,
                                                 'elements': 'str'}
            discovery_spec["discovery_type"] = {'type': 'str', 'required': True}
        else if state == deleted:
            if (self.config[0].get("delete_all") is True:
                self.validated_config = [{"delete_all": True}]
                 self.msg = "Sucessfully collected input for deletion of all the discoveries"
                 self.log(self.msg, "WARNING")
                 return self
```

Reply:
Done